### PR TITLE
Add compute node types and templates

### PR DESCRIPTION
**Pull request by emilymye**
Part 1 of https://github.com/terraform-providers/terraform-provider-google/issues/2700

Node groups to follow

-----------------------------------------------------------------
# [all]
## [terraform]
Add compute node types and templates
### [terraform-beta]
Add compute node types and templates

### CODE REVIEW

File: products/compute/api.yaml [hunk 1]
```diff
@@ -3148,6 +3148,88 @@ objects:
...
+      - !ruby/object:Api::Type::KeyValuePairs
+        name: 'nodeAffinityLabels'
+        description: |
+          Labels to use for node affinity, which will be used in
+          instance scheduling.
+      - !ruby/object:Api::Type::String
+        name: 'nodeType'
+        description: |
+          Node type to use for nodes group that are created from this template.
+          Only one of nodeTypeFlexibility and nodeType can be specified.
```
Comment (rileykarson) [inline products/compute/api.yaml]: Can we add a conflicts?

File: products/compute/terraform.yaml [hunk 2]
```diff
@@ -669,6 +669,19 @@ overrides: !ruby/object:Overrides::ResourceOverrides
...
+        name: "node_template_basic"
+        primary_resource_id: "template"
+        vars:
+          template_name: "soletenant-tmpl"
+    properties:
+      region: !ruby/object:Overrides::Terraform::PropertyOverride
+        required: false
+        default_from_api: true
+        custom_flatten: 'templates/terraform/custom_flatten/name_from_self_link.erb'
+
```
Comment (rileykarson) [inline products/compute/terraform.yaml]:
```suggestion
        description: |
          Region where nodes using this the node template will be created.
          If it is not provided, the provider region is used.
```

Comment (rileykarson) [reply products/compute/terraform.yaml]: (no idea if spacing will even almost be right)

File: templates/terraform/examples/node_template_basic.tf.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+data "google_compute_node_types" "central1a" {
+  zone = "us-central1-a"
+}
+
+resource "google_compute_node_template" "<%= ctx[:primary_resource_id] %>" {
+  name = "test-template"
```
Comment (rileykarson) [inline templates/terraform/examples/node_template_basic.tf.erb]: This should be a variable, right? Looks like you defined one in the yaml

File: products/compute/terraform.yaml (hunk 2 above)
Comment (danawillow) [reply products/compute/terraform.yaml]: I'm going to suggest on your suggestion that s/this the/this/

File: third_party/terraform/tests/data_source_google_compute_node_types_test.go [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+		}
+
+		count, ok := rs.Primary.Attributes["names.#"]
+		if !ok {
+			return errors.New("can't find 'names' attribute")
+		}
+
+		cnt, err := strconv.Atoi(count)
+		if err != nil {
+			return errors.New("failed to read number of version")
```
Comment (danawillow) [inline third_party/terraform/tests/data_source_google_compute_node_types_test.go]: nit: s/version/versions

File: third_party/terraform/website-compiled/google.erb [hunk 5]
```diff
@@ -90,6 +90,9 @@
         <a href="/docs/providers/google/d/google_project_services.html">google_project_services</a>
       </li>
       <li<%%= sidebar_current("docs-google-datasource-compute-regions") %>>
+      <a href="/docs/providers/google/d/google_compute_node_types.html">google_compute_node_types</a>
+      </li>
+      <li<%%= sidebar_current("docs-google-datasource-compute-regions") %>>
```
Comment (danawillow) [inline third_party/terraform/website-compiled/google.erb]: looks like a copy/paste typo

File: third_party/terraform/website-compiled/google.erb [hunk 6]
```diff
@@ -448,6 +451,10 @@
       <a href="/docs/providers/google/r/compute_interconnect_attachment.html">google_compute_interconnect_attachment</a>
       </li>
 
+      <li<%%= sidebar_current("docs-google-compute-project-metadata-item") %>>
```
Comment (danawillow) [inline third_party/terraform/website-compiled/google.erb]: same here

File: third_party/terraform/website/docs/d/google_compute_node_types.html.markdown [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+## Example Usage
+
+```hcl
+data "google_compute_node_types" "central1b" {
+  zone = "us-central1-b"
+}
+
+resource "google_compute_node_template" "tmpl" {
+  name               = "terraform-test-tmpl"
+  region             = "us-central1"
```
Comment (danawillow) [inline third_party/terraform/website/docs/d/google_compute_node_types.html.markdown]: nit: mind moving this equals sign and the one above it inline with the node_type one?

File: products/compute/terraform.yaml (hunk 2 above)
Comment (rileykarson) [reply products/compute/terraform.yaml]: + https://github.com/GoogleCloudPlatform/magic-modules/pull/1657/files#diff-fb4f76e7d870258668a3beac48bf164cR3190 in that case, that line was copied from there

**Pull request merged**